### ISSUE TITLE
Uncaught SyntaxError: Unexpected token P

### ISSUE THREAD

**Issue by dessant** [autocomplete-python/autocomplete-python#140]
**Atom Version**: 1.3.0
**System**: Microsoft Windows 10
**Thrown From**: [autocomplete-python](https://github.com/sadovnychyi/autocomplete-python) package, v1.2.3
### Stack Trace

Uncaught SyntaxError: Unexpected token P

```
SyntaxError: Unexpected token P
    at Object.parse (native)
    at [object Object].module.exports._deserialize (file:///C:/Users/prc/.atom/packages/autocomplete-python/lib/provider.coffee:169:23)
    at provider.BufferedProcess.stdout (file:///C:/Users/prc/.atom/packages/autocomplete-python/lib/provider.coffee:47:10)
    at Socket.<anonymous> (C:\Users\prc\AppData\Local\atom\app-1.3.0\resources\app.asar\src\buffered-process.js:85:13)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at Pipe.onread (net.js:523:20)
```
### Commands

```
     -8:28.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -7:46.3.0 editor:newline (atom-text-editor.editor.is-focused)
     -6:48.8.0 core:copy (atom-text-editor.editor.is-focused)
     -6:46.3.0 core:paste (atom-text-editor.editor.is-focused)
     -6:39.3.0 core:copy (atom-text-editor.editor.is-focused)
     -6:37.9.0 core:paste (atom-text-editor.editor.is-focused)
     -6:35.9.0 editor:newline (atom-text-editor.editor.is-focused)
     -6:05.5.0 core:copy (atom-text-editor.editor.is-focused)
  5x -0:48.2.0 core:backspace (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "packagesWithKeymapsDisabled": [
      "autocomplete-python"
    ]
  },
  "autocomplete-python": {
    "showDescriptions": false,
    "addBracketAfterFunction": true
  }
}
```

**Comment by sadovnychyi** [comment]
Does it happens all the time or just once? Can you turn on debug logs and show me some when issue occurs?

**Comment by dessant** [comment]
It happened only once, i'll turn on debug logging and report back if the error appears again.

**Comment by dessant** [comment]
It seems to be triggered by an empty response. All further requests will give the same error after that, until the process is killed and a new one started.

```
{"id": "a2aaa1d7f87ee9f25d1e7ce81a41e1d1", "results": [{"rightLabel": "", "text": "pvk", "description": "", "type": "forstmt"}]}
{"id": "131f6035bca67fc4fefcb9e14af9f2ef", "results": [{"rightLabel": "", "text": "pvk", "description": "", "type": "forstmt"}]}
{"id": "3e5d6bfb12bb2ee518861112b8688d5b", "results": []}

/C:/Users/prc/.atom/packages/autocomplete-python/lib/log.coffee:10 autocomplete-python: Got 9 lines
index.html:1 Uncaught SyntaxError: Unexpected token P
```

**Comment by sadovnychyi** [comment]
Can you try to reproduce it with certain steps?

I guess this is something windows-specific, and I'm not really comfortable with testing under windows. But if you are sure that I can reproduce it – I'll do it.

**Comment by sadovnychyi** [comment]
Assuming it didn't happen anymore.

**Comment by dessant** [comment]
It does happen sometimes, but i don't know how to further help with debugging. I dindn't see any commit that would have made the bug go away either.

**Comment by dessant** [comment]
I'm returning `[]` in _deserialize, but nothing so far.
Btw, I have the fuzzy matcher turned off, yet I start getting fuzzy matches when there are no exact ones found, is this expected?

![capture](https://cloud.githubusercontent.com/assets/7513068/11912478/c08af2de-a646-11e5-9c03-27ad915ccbaa.PNG)

**Comment by sadovnychyi** [comment]
Hopefully since #147 is fixed you can catch more info about this.

**Comment by sadovnychyi** [comment]
Any updates on this?

**Comment by dessant** [comment]
The logs are the same when it happens, i guess you'd have to use atom on windows for a while to trigger the bug.

**Comment by sadovnychyi** [comment]
Should be fixed in 1.8.9.
